pysph/sph/wc/crksph.py

```diff
@@ -804,73 +977,163 @@ def configure_solver(self, kernel=None, integrator_cls=None,
         )
 
     def get_equations(self):
-        from pysph.sph.basic_equations import XSPHCorrection
-        from pysph.sph.wc.basic import TaitEOS
         from pysph.sph.wc.viscosity import LaminarViscosity
         all = self.fluids
 
-        equations = []
+        equations_stage1 = []
+        equations_stage2 = []
+
+        eq00 = []
+        for fluid in self.fluids:
+            eq00.append(
+                StateEquation(dest=fluid, sources=None, gamma=self.gamma)
+            )
+            eq00.append(
+                SpeedOfSound(dest=fluid, sources=None, gamma=self.gamma)
+            )
+        equations_stage1.append(Group(equations=eq00))
+
+        gh = []
+        for fluid in self.fluids:
+            gh.append(
```

Comment:
Same here, if there is no ghost this should not be done, right?  Unless this is safe to use and we are guaranteed that orig_idx is correctly set?